### PR TITLE
Added AWS-AHUG alinux2 pipeline

### PR DESCRIPTION
**Pull request by bollig**
Testing new pipeline for ARM (alinux2)

### CODE REVIEW

**Comment by bollig** [comment]
Test rebuild from comments

**Comment by bollig** [comment]
will squash once pipeline builds

File: share/spack/gitlab/cloud_pipelines/stacks/aws-ahug/spack.yaml [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+    #  variants: +pmix sysconfdir=/opt/slurm/etc
+    trilinos:
+      variants: +amesos +amesos2 +anasazi +aztec +belos +boost +epetra +epetraext +ifpack +ifpack2 +intrepid +intrepid2 +isorropia +kokkos +ml +minitensor +muelu +nox +piro +phalanx +rol +rythmos +sacado +stk +shards +shylu +stokhos +stratimikos +teko +tempus +tpetra +trilinoscouplings +zoltan +zoltan2 +superlu-dist gotype=long_long
+    xz:
+      variants: +pic
+
+  definitions:
+
+  - compiler_specs:
+    - gcc
```
Comment (scottwittenburg) [inline share/spack/gitlab/cloud_pipelines/stacks/aws-ahug/spack.yaml:73]: Version?

File: share/spack/gitlab/cloud_pipelines/stacks/aws-ahug-aarch64/spack.yaml [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+    #  variants: +pmix sysconfdir=/opt/slurm/etc
+    trilinos:
+      variants: +amesos +amesos2 +anasazi +aztec +belos +boost +epetra +epetraext +ifpack +ifpack2 +intrepid +intrepid2 +isorropia +kokkos +ml +minitensor +muelu +nox +piro +phalanx +rol +rythmos +sacado +stk +shards +shylu +stokhos +stratimikos +teko +tempus +tpetra +trilinoscouplings +zoltan +zoltan2 +superlu-dist gotype=long_long
+    xz:
+      variants: +pic
+
+  definitions:
+
+  - compiler_specs:
+    - gcc
```
Comment (scottwittenburg) [inline share/spack/gitlab/cloud_pipelines/stacks/aws-ahug-aarch64/spack.yaml:73]: Version here?

File: share/spack/gitlab/cloud_pipelines/stacks/aws-ahug-aarch64/spack.yaml [hunk 3]
```diff
@@ -0,0 +1,321 @@
+# TODO:
```
Comment (scottwittenburg) [inline share/spack/gitlab/cloud_pipelines/stacks/aws-ahug-aarch64/spack.yaml]: Are these comments still relevant, i.e. to be done?  If resolved, can they be removed?

File: share/spack/gitlab/cloud_pipelines/stacks/aws-ahug/spack.yaml [hunk 4]
```diff
@@ -0,0 +1,322 @@
+# TODO:
```
Comment (scottwittenburg) [inline share/spack/gitlab/cloud_pipelines/stacks/aws-ahug/spack.yaml]: Still todo?  Or can these comments be removed?

**Review by scottwittenburg — CHANGES_REQUESTED** [review]
This looks good, thanks @bollig, just a couple questions/requests.

File: share/spack/gitlab/cloud_pipelines/stacks/aws-ahug-aarch64/spack.yaml (hunk 2 above)
Comment (bollig) [reply share/spack/gitlab/cloud_pipelines/stacks/aws-ahug-aarch64/spack.yaml:73]: In this case, gcc is the package that is built. I'd prefer the pipeline to track the latest version and continuously rebuild as new versions and hashes are populated. The `compiler:` block governs which version of gcc (gcc@7.3.1) is used to build packages.

Comment (scottwittenburg) [reply share/spack/gitlab/cloud_pipelines/stacks/aws-ahug-aarch64/spack.yaml:73]: Yeah, ok, that makes sense.

**Review by scottwittenburg — DISMISSED** [review]
Ok looks good, thanks @bollig :+1:

**Comment by bollig** [comment]
Looks like this pipeline is failing on the same issues resolved by #30798

**Review by scottwittenburg — DISMISSED** [review]
LGTM pending CI

**Comment by bollig** [comment]
@spackbot re-run pipeline

**Comment by scottwittenburg** [comment]
@spackbot run pipeline

**Review by scottwittenburg — APPROVED** [review]

**Pull request merged**